app/src/main/java/com/example/habitsmasher/ForgotPasswordDialog.java

```diff
@@ -0,0 +1,71 @@
+package com.example.habitsmasher;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+
+import com.google.firebase.auth.FirebaseAuth;
+
+
+/**
+ * This class is the dialog that opens when the user forgets their password.
+ * @author Jacob Nguyen
+ */
+public class ForgotPasswordDialog extends DialogFragment {
+
+    private static final String EMAIL_SENT_MESSAGE = "Email sent, please check your email";
+
+    FirebaseAuth _auth;
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.forgot_user_password_dialog, container, false);
+
+        _auth = FirebaseAuth.getInstance();
+
+        EditText forgotPasswordEdittext = view.findViewById(R.id.reset_password_dialog);
+        Button forgotPasswordButton = view.findViewById(R.id.reset_password_button);
+
+        setClickListenerForConfirmButton(forgotPasswordEdittext, forgotPasswordButton);
+
+        return view;
+    }
+
+    /**
+     * This method listens for when the user presses the confirm button to get an email sent to them to reset their email.
+     * @param emailInput email inputted by user
+     * @param confirmButton button user presses to get email sent to them
+     */
+    private void setClickListenerForConfirmButton(EditText emailInput, Button confirmButton) {
+        confirmButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                UserValidator userValidator = new UserValidator(emailInput);
```

Comment:
Not sure if we should be using the user validator for this, since at the moment it's only checking the format of the email. Perhaps we should use a new EmailVaildator class.

Reply:
I mean it's not the end of the world but you do have a point, it could be its own class.

Reply:
Actually it wouldn't be too hard to make it its own class